### PR TITLE
add Terraform Test to module

### PR DESCRIPTION
**Pull request by drewmullen**
provide acceptance tests using `terraform test`. This conflicts with #3 (proposed)

### Test Scenarios:
1. validates that `tags` come back in the `awscc` format regardless of the given format and `tags_aws` come back in the `aws` format, regardless of the given format.
2. validates tags work with resources (`aws_vpc` & `awscc_ec2_vpc`)
3. validates account names are mapped correctly 

Ive removed the `internal/` but we can discuss if theres value creating a test that actually makes the same `iam` resources. 

### Note
`terraform test` requires the module references be:
1. in the same dir as the tests
2. `source = "../.."`
these are currently [hard requirements](https://discuss.hashicorp.com/t/terraform-test-should-be-able-to-reference-hcl-config-from-example-dirs/32553/3). I was able to get around them by creating symlinks to our examples hcl config files

### CODE REVIEW

File: tests/examples_formatted_tags/test.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    test_awscc_to_awscc = tolist(module.awscc_labels.tags)
+  }
+}
+
+resource "test_assertions" "matching_formatted_labels_awscc" {
+  component = "tags"
+
+  check "valid_tags_for_awscc" {
+    description = "Validate tags formatting for AWSCC Provider usage. expected: {key=\"foo\", values=\"bar\"}"
+    condition   = length(setsubtract(local.awscc_tags_formatted["test_aws_to_awscc"], local.awscc_tags_formatted["test_awscc_to_awscc"])) == 0
```
[the remark this answers is not in the document]
Comment (drewmullen) [reply tests/examples_formatted_tags/test.tf]: yes this is a good point. was only checking 1 direction 😬

**Comment by andrew-glenn** [comment]
Note: https://github.com/hashicorp/terraform/issues/30071 is relevant.

**Review by andrew-glenn — CHANGES_REQUESTED** [review]
Preventing merge for now, due to underlying issue discovered regarding `terraform test` and implicit dependencies.

**Comment by drewmullen** [comment]
Closing - pending future releases of `terraform test` we can re-evaluate

**Pull request closed without merging**